src/sensors/sensor_poller/star_tracker_poller/star_tracker_poller.c

```diff
@@ -223,22 +224,52 @@ static void irisc_tetra(float st_return[]) {
     strcat(exe_path,"usr/local/astrometry/bin/solve-field");
     */
     strcpy(st_img_path, get_top_dir());
-    strcat(st_img_path,"output/guiding/star_tracker/st_img.fit");
+    strcat(st_img_path,"output/guiding/star_tracker/st_img.fit ");
+
+    /* lisFlag, 1 if we are completely lost in space. 0 for only slightly. */
+    static int lisFlag = 1;
+    static char* oldRa[2] = "0";
+    static char* oldDec[2]; = "0";
+    char* stRad = "15";
+    
     
-    char* cmd[7] = {
+    char* cmd[8] = {
         "chrt",
         "-f",
         "23",
         "/usr/local/astrometry/bin/solve-field",
-        "-pO",
+        "-pOo",
+        "none",
+        "--scale-low",
+        "1",
         st_img_path,
         NULL
     };
 
+    char* raDecFlags[7] = {
+        "--ra",
+        raCurr,
+        "--dec",
+        decCurr,
+        "--radius",
+        stRad,
+        NULL 
+    };
+
+    char* raDecCmd[16];
+
     int out_fd = -1;
 
     st_running = 1;
-    py_pid = popen2(cmd, NULL, &out_fd);
+
+    if (!lisFlag) {
+        strcpy(raDecCmd, cmd);
```

Comment:
strcpy will stop at `'\0'`, and each string literal ends with `'\0'`.

It would probably be easier to have `cmd` and `raDecFlags` in the same `char* []` and replace the pointer after `stRad` with `NULL` if the ra & dec values shouldn't be used.